core/services/relay/evm/method_binding.go

```diff
@@ -62,5 +65,8 @@ func (m *methodBinding) GetLatestValue(ctx context.Context, params, returnValue
 func (m *methodBinding) Bind(ctx context.Context, binding commontypes.BoundContract) error {
 	m.address = common.HexToAddress(binding.Address)
 	m.bound = true
+	if binding.Pending {
+		m.blockNumber = big.NewInt(rpc.PendingBlockNumber.Int64())
+	}
```

Comment:
The purpose of this flag is to differentiate between unconfirmed blocks and finalized blocks. It was originally named `Finalized` but due to miscommunications during the review process of the CLIP it ended up getting renamed to Pending.

There was never any intention to support reading data from pending blocks, and there is currently no easy way to support that for events.

I agree that we shouldn't be ignoring it here, and that the name of it is misaligned with go-ethereum naming, but making this change here would not bring things into alignment because there would be no easy way to make a similar change in the event binding code, where `Pending=true` means `Latest`:
https://github.com/smartcontractkit/chainlink-common/pull/444

If we were keeping it a boolean, there would be a very simple way to bring everything into alignment which is to rename it back to `Finalized` and pass `rpc.FinalizedBlockNumber.Int64()` here if it's true. 

However, there have since been 3 reasons why we've decided it needs to change to an integer instead of a boolean anyway.  The first was a request from mgmt that we don't assume deterministic finality and allow plugins to use confidence levels instead of just finalized vs not finalized.   The second was the realization that CCIP has a requirement already for partial confidence levels (although that one I think we might be able to get away with only supporting in the case of `QueryOne`, rather than GetLatestValue?)  And the third was finding out that Median  uses `MinConfirmations=1` rather than `MinConfirmations=0` which correspond to LatestBlock. Presently we're passing `LatestBlock` for the ChainReader impl of `LatestRoundReported` but that's a change in functionality from the legacy implementation, so if we left it as a boolean there would be more work to be done in validating that this change wouldn't cause any issues for node ops. 

None of the 3 by themselves is necessarily definitive, but I think all 3 together make for a pretty good case that we should change this, or we'll keep running into more issues.

I think the best way forward here is to do something like this, where we allow both positive & negative values: positive values can be a confidence level while negative values could have special meaning, such as reading from pending block:
https://github.com/smartcontractkit/chainlink-common/pull/444